core/src/main/java/io/questdb/griffin/DatabaseSnapshotAgent.java

```diff
@@ -67,102 +82,158 @@ public static void recoverSnapshot(CairoEngine engine) {
-                    copyPath.trimTo(cplen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
-                    if (ff.exists(path) && ff.exists(copyPath)) {
-                        if (ff.copy(path, copyPath) < 0) {
+                    srcPath.trimTo(srcPathLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
+                    dstPath.trimTo(dstPathLen).concat(TableUtils.COLUMN_VERSION_FILE_NAME).$();
+                    if (ff.exists(srcPath) && ff.exists(dstPath)) {
+                        if (ff.copy(srcPath, dstPath) < 0) {
                             LOG.error()
-                                    .$("could not copy _cv file [src=").$(path)
-                                    .$(", dst=").$(copyPath)
+                                    .$("could not copy _cv file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
                                     .$(", errno=").$(ff.errno())
                                     .$(']').$();
                         } else {
                             recoveredCVFiles.incrementAndGet();
                             LOG.info()
-                                    .$("recovered _cv file [src=").$(path)
-                                    .$(", dst=").$(copyPath)
+                                    .$("recovered _cv file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
+                                    .$(']').$();
+                        }
+                    }
+
+                    // Go inside SEQ_DIR
+                    srcPath.trimTo(srcPathLen).concat(WalUtils.SEQ_DIR);
+                    srcPathLen = srcPath.length();
+                    srcPath.concat(TableUtils.META_FILE_NAME).$();
+
+                    dstPath.trimTo(dstPathLen).concat(WalUtils.SEQ_DIR);
+                    dstPathLen = dstPath.length();
+                    dstPath.concat(TableUtils.META_FILE_NAME).$();
+
+                    if (ff.exists(srcPath) && ff.exists(dstPath)) {
+                        if (ff.copy(srcPath, dstPath) < 0) {
+                            LOG.error()
+                                    .$("could not copy ").$(TableUtils.META_FILE_NAME).$(" file [src=").$(srcPath)
+                                    .$(", dst=").$(dstPath)
+                                    .$(", errno=").$(ff.errno())
+                                    .$(']').$();
+                        } else {
+                            try {
+                                srcPath.trimTo(srcPathLen);
+                                openSmallFile(ff, srcPath, srcPathLen, memFile, TableUtils.TXN_FILE_NAME, MemoryTag.MMAP_TX_LOG);
+                                long newMaxTxn = memFile.getLong(0L);
+
+                                memFile.smallFile(ff, dstPath, MemoryTag.MMAP_SEQUENCER_METADATA);
+                                long maxStructureVersion = memFile.getLong(SEQ_META_OFFSET_STRUCTURE_VERSION);
```

Comment:
This must match `txn_seq\_meta` structure version, should be overwritten in this file instead of reading it from here

Reply:
I didn't get what you mean.
`maxStructureVersion` and `txnMetaMemSize` are read from `txn_seq/_meta`. Should we patch `txn_seq/_meta` instead?

Reply:
Add a comment please of what file is read /written relative to the table